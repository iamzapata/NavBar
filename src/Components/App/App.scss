@charset "utf-8";
@import "~bulma/bulma";
@import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:400,700");

html {
  font-family: "Source Sans Pro", sans-serif;
  min-height: 100v;
  display: flex;
  align-items: center;
  justify-content: center;
}
