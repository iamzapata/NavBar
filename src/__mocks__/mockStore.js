export default {
  Theme: { themeName: "Pinterest" },
  NewsFeed: { feed: [], isLoading: false },
  UserProfile: {
    profile: { bio: "", interests: "", first_name: "", last_name: "" },
    isLoading: false
  },
  UpcomingEvents: { events: [], isLoading: false }
}
