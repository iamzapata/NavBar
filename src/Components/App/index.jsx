import React from "react"
import { Route, NavLink } from "react-router-dom"
import NewsFeed from "Components/Pages/NewsFeed"
import People from "Components/Pages/People"
import Profile from "Components/Pages/Profile"
import "./App.scss"

const App = () => (
  <div className="App">
    <ul>
      <li>
        <NavLink to="/feed">News</NavLink>
      </li>
      <li>
        <NavLink to="/people">People</NavLink>
      </li>
      <li>Food</li>
      <li>Admin</li>
      <li>More</li>
    </ul>
    <Route path="/feed" component={NewsFeed} />
    <Route path="/people" component={People} />
  </div>
)

export default App
