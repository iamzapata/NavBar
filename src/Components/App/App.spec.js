import React from "react"
import { render, cleanup } from "react-testing-library"
import { BrowserRouter as Router } from "react-router-dom"
import { Provider } from "react-redux"
import ReduxThunk from "redux-thunk"
import configureMockStore from "redux-mock-store"
import mockStore from "__mocks__/mockStore.js"
import App from "./index"

const configureStore = configureMockStore([ReduxThunk])
const store = configureStore(mockStore)

describe("<App />", () => {
  afterEach(cleanup)
  it("Should render self", () => {
    const { container } = render(
      <Router>
        <Provider store={store}>
          <App />
        </Provider>
      </Router>
    )
    expect(container.firstChild).toBeTruthy()
    expect(container.firstChild).toMatchSnapshot()
  })
})
