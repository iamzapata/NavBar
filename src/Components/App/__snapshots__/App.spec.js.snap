// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Should render self 1`] = `
<div
  class="App Pinterest__Theme"
>
  <nav
    class="NavBar navbar"
  >
    <div
      class="navbar-brand"
    >
      <a
        aria-current="page"
        class="NavLInkBrand navbar-item active"
        href="/"
      >
        <img
          alt="Pinterest"
          src="test-file-stub"
        />
        <span
          class="BrandName is-size-5"
        >
          Pinterest
        </span>
      </a>
      <button
        class="navbar-burger burger"
        type="button"
      >
        <span />
        <span />
        <span />
      </button>
    </div>
    <div
      class="navbar-menu"
    >
      <div
        class="navbar-start"
        role="button"
        tabindex="0"
      >
        <a
          class="navbar-item"
          href="/feed"
        >
          News
        </a>
        <a
          class="navbar-item"
          href="/people"
        >
          People
        </a>
        <a
          class="navbar-item"
          href="/food"
        >
          Food
        </a>
        <a
          class="navbar-item"
          href="/admin"
        >
          Admin
        </a>
        <a
          class="navbar-item"
          href="/more"
        >
          More
        </a>
        <div
          class="SearchBar control has-icons-left navbar-item"
        >
          <input
            class="input"
            placeholder="Find a Pintrest"
            type="text"
          />
          <span
            class="icon is-small is-left"
          >
            <i
              class="fa fa fa-search"
            />
          </span>
        </div>
      </div>
      <div
        class="navbar-end"
      >
        <div
          class="NotificationsTray navbar-item"
        >
          <button
            class="button"
            type="button"
          >
            <i
              class="fa fa-bell"
            />
            <span
              class="Notifications__Count"
            >
              7
              +
            </span>
          </button>
        </div>
        <div
          class="ProfileDropdown dropdown navbar-item"
        >
          <div
            class="dropdown-trigger"
          >
            <button
              aria-controls="ProfileDropdown"
              aria-haspopup="true"
              class="button"
              data-testid="ProfileButton"
              type="button"
            >
              <img
                alt="Profile"
                class="image"
                src="test-file-stub"
              />
              <span
                class="icon is-small"
              >
                <i
                  aria-hidden="true"
                  class="fa fa-caret-down"
                />
              </span>
            </button>
          </div>
          <div
            class="dropdown-menu"
            id="ProfileDropdown"
            role="menu"
          >
            <div
              class="dropdown-content"
              role="button"
              tabindex="0"
            >
              <a
                class="dropdown-item"
                href="/profile/1"
              >
                Profile
              </a>
              <a
                class="dropdown-item"
                href="/settings"
              >
                Settings
              </a>
              <a
                class="dropdown-item"
                href="/sign-out"
              >
                Sign Out
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <div
    class="PageContainer container"
    style="opacity: 0; visibility: hidden;"
  >
    <div
      class="columns"
    >
      <div
        class="column"
      />
      <div
        class="column is-3"
      >
        <div
          class="UserSummary"
        >
          <div
            class="card"
          >
            <div
              class="card-content"
            >
              <div
                class="media"
              >
                <div
                  class="media-left"
                >
                  <figure
                    class="image is-48x48"
                  >
                    <img
                      alt="User"
                    />
                  </figure>
                </div>
                <div
                  class="media-content"
                >
                  <p
                    class="title is-size-4"
                  >
                     
                  </p>
                  <p
                    class="subtitle is-size-6"
                  />
                </div>
              </div>
              <div
                class="content is-size-6"
              />
              <hr />
            </div>
          </div>
        </div>
        <div
          class="UpcomingEvents"
        >
          <span
            class="is-size-7"
          >
            UPCOMING EVENTS
          </span>
        </div>
      </div>
      <div
        class="FeedList column is-7"
      >
        <p
          class="FeedList__Header"
        >
          LATEST
        </p>
        <div
          class="FeedList__Items"
        />
      </div>
      <div
        class="column"
      />
    </div>
    <div
      class="sk-fading-circle"
      style="width: 50px; height: 50px; display: none;"
    >
      <div
        class="sk-circle1 sk-circle"
      />
      <div
        class="sk-circle2 sk-circle"
      />
      <div
        class="sk-circle3 sk-circle"
      />
      <div
        class="sk-circle4 sk-circle"
      />
      <div
        class="sk-circle5 sk-circle"
      />
      <div
        class="sk-circle6 sk-circle"
      />
      <div
        class="sk-circle7 sk-circle"
      />
      <div
        class="sk-circle8 sk-circle"
      />
      <div
        class="sk-circle9 sk-circle"
      />
      <div
        class="sk-circle10 sk-circle"
      />
      <div
        class="sk-circle11 sk-circle"
      />
      <div
        class="sk-circle12 sk-circle"
      />
    </div>
  </div>
  <footer>
    <div
      class="ThemeSwitcher field Pinterest"
    >
      <input
        checked=""
        class="switch is-rounded"
        id="themeSwitcherButton"
        type="checkbox"
      />
      <label
        for="themeSwitcherButton"
      />
    </div>
  </footer>
</div>
`;
